iocore/net/UnixNet.cc

```diff
@@ -429,25 +439,43 @@ NetHandler::process_ready_list()
       }
 #endif
     }
+    // Must clear in case the debug enable changes
+    if (set_flag) {
+      set_cont_flags(0);
+    }
   }
 #else  /* !USE_EDGE_TRIGGER */
   while ((vc = read_ready_list.dequeue())) {
-    diags->set_override(vc->control.debug_override);
+    bool set_flag = vc->control_flags.is_set();
+    if (set_flag) {
+      set_cont_flags(vc->control_flags);
+    }
```

Comment:
I added these changes based on @bryancall's request that we don't set the debug control thread local storage if we aren't debugging.  In my opinion this solution is awkward, and based on our experience running this logic in our 5.3 and 7.1 the thread local storage write hasn't been a performance bottleneck.  We trade two boolean checks for one thread local storage write.

I did update the PR to use the C++ thread local storage nomenclature which I do like.  It simplifies the code.

Reply:
I ran a benchmark on thread_local to see how it compares to doing 2 boolean checks and there is little to no difference in performance.  Both were doing around 1 billion in 3.4 to 3.5 seconds on my Mac.  Glad so see that the code was changed to thread_local and you can revert the bool checks if you would like.